report/src/main/java/com/megaease/easeagent/report/async/zipkin/AgentByteBoundedQueue.java

```diff
@@ -14,126 +14,97 @@
-            return true;
-        } finally {
-            lock.unlock();
+        if (maxSize == queue.size()) {
+            return false;
         }
+        if (sizeInBytes.updateAndGet(pre -> pre + nextSizeInBytes) > maxBytes) {
+            return false;
+        }
+        queue.offer(new DataWrapper<>(next, nextSizeInBytes));
+        return true;
+    }
+
+    int doDrain(WithSizeConsumer<S> consumer, DataWrapper<S> firstPoll) {
+        int drainedCount = 0;
+        int drainedSizeInBytes = 0;
+        DataWrapper<S> next = firstPoll;
+        do {
+            int nextSizeInBytes = next.getSizeInBytes();
+            if (consumer.offer(next.getElement(), nextSizeInBytes)) {
+                drainedCount++;
+                drainedSizeInBytes += nextSizeInBytes;
+            } else {
+                break;
+            }
+            next = queue.poll();
+            if (next == null) break;
+        } while (drainedCount < queue.size());
+        final int updateValue = drainedSizeInBytes;
+        sizeInBytes.updateAndGet(pre -> pre - updateValue);
+        return drainedCount;
     }
 
-    /** Blocks for up to nanosTimeout for spans to appear. Then, consume as many as possible. */
     public int drainTo(WithSizeConsumer<S> consumer, long nanosTimeout) {
+        DataWrapper<S> firstPoll;
         try {
-            // This may be called by multiple threads. If one is holding a lock, another is waiting. We
-            // use lockInterruptibly to ensure the one waiting can be interrupted.
-            lock.lockInterruptibly();
-            try {
-                long nanosLeft = nanosTimeout;
-                while (count == 0) {
-                    if (nanosLeft <= 0) return 0;
-                    nanosLeft = available.awaitNanos(nanosLeft);
-                }
-                return doDrain(consumer);
-            } finally {
-                lock.unlock();
-            }
+            firstPoll = queue.poll(nanosTimeout, TimeUnit.NANOSECONDS);
         } catch (InterruptedException e) {
-            // use lockInterruptibly to ensure the one waiting can be interrupted.
             return 0;
         }
+        if (firstPoll == null) {
+            return 0;
+        }
+        return doDrain(consumer, firstPoll);
     }
 
     public int getCount() {
-        return count;
+        return queue.size();
     }
 
     public int getSizeInBytes() {
-        return sizeInBytes;
+        return sizeInBytes.intValue();
     }
 
-    /** Clears the queue unconditionally and returns count of spans cleared. */
     public int clear() {
-        lock.lock();
-        try {
-            int result = count;
-            count = sizeInBytes = readPos = writePos = 0;
-            Arrays.fill(elements, null);
-            return result;
-        } finally {
-            lock.unlock();
-        }
+        int result = queue.size();
+        queue.clear();
+        sizeInBytes.set(0L);
+        return result;
```

Comment:
This modified class has an atomic operation vulnerability, I hope you can optimize it, especially the operation here